Feature/search screen design

Belal-Alkhatib opened the pull request:
# Pull Request Template

## Description
This PR introduces the initial implementation of the search screen UI and refines several shared UI components to support its layout and interactions.

---

## Changes Made
List of changes introduced in this pull request:

### ✨ Features:
- Implemented `SearchScreen.kt` containing:
  - `LazyColumn` with scrollable content
  - Sticky `DefaultAppBar` and `TextField` for input
- Created `SuggestionsHubSection.kt` to display suggested searches (e.g., World Tour, Find by Actor)
- Added `RecentSearchesSection.kt` to show previous search terms
- New `RecentSearchItem.kt` composable in `designSystem` for consistent UI

### 🧠 State Management:
- Added `SearchUiState.kt` and `RecentSearchItemUiState.kt` in the `viewModel` layer
- Introduced `isSearchQueryEmpty` to control the display logic of suggestion and recent sections

### 🛠 Component Updates:
- **TextField**: Now accepts a `modifier` for better layout control
- **TopAppBar**: Adjusted padding for better alignment
- **GlobalSearchHub**:
  - New `onItemClick` callback
  - Set `aspectRatio` to `1.6f`
  - `fillMaxWidth` applied
  - Made `clickable` for interaction support

### 🎨 UI/UX Improvements:
- Removed global horizontal padding from `LazyColumn` in `SearchScreen`
- Applied individual padding to top-level elements (`AppBar`, `TextField`, `SuggestionsHub`, `RecentSearchesSection`)
- Updated `TabLayout` to wrap content in a `Column` and placed `HorizontalDivider` below `ScrollableTabRow`

---

## Screenshots (if applicable)
Include screenshots or GIFs to showcase the changes, especially if they impact the UI.

---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

Belal-Alkhatib committed:
Refactor: update foundation layout dependency

Belal-Alkhatib committed:
feat: add clock icon and search suggestions string

Belal-Alkhatib committed:
fix: update TextField, GlobalSearchHub, and TopAppBar components

- Modify TextField to accept a modifier.
- Adjust padding in TopAppBar.
- Enhance GlobalSearchHub:
    - Add onItemClick callback.
    - Set aspectRatio to 1.6f.
    - Make the component fillMaxWidth.
    - Make the component clickable.

Belal-Alkhatib committed:
feat: implement initial search screen UI

This commit introduces the basic structure and UI elements for the search screen.

Key changes include:
- Added `SearchScreen.kt` with a `LazyColumn` for scrollable content.
- Implemented `SearchContent` composable containing:
    - `DefaultAppBar` for the screen title.
    - A sticky `TextField` for search input.
- Created `SuggestionsHubSection.kt` to display search suggestion cards (World Tour, Find by Actor).
- Added `RecentSearchesSection.kt` to display a list of recent search items.
- Introduced `RecentSearchItem.kt` composable in `designSystem` for individual recent search entries.
- Defined `SearchUiState.kt` and `RecentSearchItemUiState.kt` in `viewModel` to represent the UI state for the search screen and recent searches respectively.

Belal-Alkhatib committed:
Clean up: Remove unused `Color` import

Belal-Alkhatib committed:
Fix: remove horizontal padding from search screen and add padding to items

This commit adjusts the padding in the SearchScreen and its child components.
The main horizontal padding has been removed from the `LazyColumn` in `SearchScreen`.
Instead, padding has been applied individually to the `DefaultAppBar`, `TextField`, `SuggestionsHub`, and `RecentSearchesSection`.

Additionally, a new field `isSearchQueryEmpty` has been added to `SearchUiState` to control the visibility of suggestions and recent searches.
The `TabLayout` now wraps its content in a `Column` and places the `HorizontalDivider` below the `ScrollableTabRow`.

ThoriaHamdy99 commented on designSystem/src/main/java/com/example/designsystem/components/RecentSearchItem.kt:
```diff
@@ -0,0 +1,61 @@
+@Composable
+fun RecentSearchItem(
+    title: String,
+    onCancelClick : () -> Unit,
```
I think these clicks will be optional so we can add default values on it --> {}

Belal-Alkhatib committed:
Refactor: Simplify RecentSearchesSection and update SearchScreen logic

This commit refactors the `RecentSearchesSection` and related components:

- `RecentSearchesSection`:
    - Now accepts a `List<String>` for `recentSearchItems` instead of `List<RecentSearchItemUiState>`.
    - A `HorizontalDivider` is now added between `RecentSearchItem`s within the `items` block if it's not the last item.
- `RecentSearchItem`:
    - Removed `isLastItemInList` parameter as the divider logic is now handled in `RecentSearchesSection`.
    - Horizontal padding is now applied directly to the `Row` within `RecentSearchItem`.
- `SearchScreen`:
    - The visibility of `SuggestionsHubSection` and `RecentSearchesSection` is now controlled by checking if `state.query.isEmpty()`.
    - `TabsLayout` visibility is controlled by `!state.query.isEmpty()`.
    - `TabsLayout` now uses `state.selectedTabOption.index` for `selectedIndex`.
    - Dummy `RecentSearchItemUiState` list has been removed.
    - `RecentSearchesSection` is now called with an empty list.

nabilBouzineDev commented:
Please fix the issues we discussed on the meeting for `RecentSearchItem`

Belal-Alkhatib committed:
feature: define UI state for search, media items, genres, and tabs

nabilBouzineDev committed:
feature: add  custom exception hierarchy for the Aflami application.

nabilBouzineDev committed:
feature: add `BaseViewModel` class that provides common functionality for ViewModels.

Belal-Alkhatib committed:
feature: add loading and error states to SearchUiState

Belal-Alkhatib committed:
feature: add loading and error states to SearchUiState

Belal-Alkhatib committed:
Merge branch 'feature/search-error-ui-state' into feature/search-screen-design

Belal-Alkhatib committed:
Merge branch 'feature/implement-search' into feature/search-screen-design

# Conflicts:
#	ui/build.gradle.kts

nabilBouzineDev committed:
Merge branch 'feature/implement-search' of github.com:Amsterdam-Team/Aflami into feature/search-screen-design

# Conflicts:
#	ui/build.gradle.kts

nabilBouzineDev committed:
chore: remove kotlinx-datetime dependency from entity module

nabilBouzineDev committed:
Merge branch 'feature/implement-search' into feature/search-screen-design

# Conflicts:
#	ui/build.gradle.kts

nabilBouzineDev committed:
chore: remove redundant foundation-layout dependency

nabilBouzineDev committed:
feature: add search use cases

nabilBouzineDev committed:
feature: add recent search use cases

nabilBouzineDev committed:
refactor: rename SearchViewModel to GlobalSearchViewModel and add search interaction listeners

saraashraf11 commented:
resolve your conflicts plz

Belal-Alkhatib committed:
Merge branch 'feature/implement-search' into feature/search-screen-design

# Conflicts:
#	viewModel/src/main/java/com/example/viewmodel/BaseViewModel.kt

Belal-Alkhatib committed:
Merge remote-tracking branch 'origin/feature/search-screen-design' into feature/search-screen-design

nabilBouzineDev committed:
refactor: update rating and categoryName parameters to be optional.

tasneem-hakeem committed:
Merge remote-tracking branch 'origin/feature/implement-search' into feature/search-screen-design

# Conflicts:
#	viewModel/build.gradle.kts
#	viewModel/src/main/java/com/example/viewmodel/BaseViewModel.kt

tasneem-hakeem committed:
refactor: update search UI state and effect

tasneem-hakeem committed:
refactor: update package name

tasneem-hakeem committed:
fix: delete repeated AflamiException file

tasneem-hakeem committed:
refactor: change rating data type in GetTvShowByKeywordUseCase

tasneem-hakeem committed:
feature: add unknown error state to SearchErrorUiState

tasneem-hakeem committed:
feature: implement GlobalSearchViewModel with filter and recent search functionality

Belal-Alkhatib committed:
fix: usecases import path

Belal-Alkhatib committed:
refactor: connect SearchScreen to ViewModel and handle UI effects

Belal-Alkhatib committed:
Merge branch 'feature/implement-search' into feature/search-screen-design

Belal-Alkhatib committed:
Merge branch 'feature/search-screen-design' into feature/add-search-events

Belal-Alkhatib committed:
fix: correct package name for use cases

Belal-Alkhatib committed:
feature: add GlobalSearchViewModel and related use cases to Koin modules

Belal-Alkhatib committed:
Merge branch 'refs/heads/feature/implement-search' into feature/add-search-events

Belal-Alkhatib committed:
refactor: Enhance SearchScreen with animated sections and real-time updates

Belal-Alkhatib committed:
refactor: improve GlobalSearchViewModel logic and structure

tasneem-hakeem committed:
feature: add movie and tv show category data sources and repository

tasneem-hakeem committed:
feature: update FilterDialog with new state and interactions

nabilBouzineDev committed:
refactor: fix package name inconsistencies

nabilBouzineDev committed:
refactor: update genre names and icons in filter dialog

nabilBouzineDev committed:
Merge branch 'feature/implement-search' of github.com:Amsterdam-Team/Aflami into feature/search-screen-design

# Conflicts:
#	app/src/main/java/com/amsterdam/aflami/di/useCaceModule.kt
#	app/src/main/java/com/amsterdam/aflami/di/viewModelModule.kt
#	domain/src/main/java/com/example/domain/useCase/GetMoviesByActorUseCase.kt
#	localDatasource/src/main/java/com/example/localdatasource/roomDataBase/AflamiDatabase.kt
#	localDatasource/src/main/java/com/example/localdatasource/roomDataBase/daos/MovieDao.kt
#	repository/src/main/java/com/example/repository/dto/local/LocalMovieCategoryDto.kt
#	repository/src/main/java/com/example/repository/dto/local/relation/MovieWithCategories.kt
#	repository/src/main/java/com/example/repository/mapper/local/CategoryLocalMapper.kt

nabilBouzineDev committed:
fix: solve merge conflicts relate to local dtos, di and MovieRepositoryImpl.kt

nabilBouzineDev committed:
Merge branch 'feature/implement-search' of github.com:Amsterdam-Team/Aflami into feature/search-screen-design

# Conflicts:
#	ui/src/main/res/values/strings.xml
